protos/orchestrator_service.proto

```diff
@@ -376,6 +376,124 @@ message DeleteTaskHubResponse {
     //no playload
 }
 
+message SignalEntityRequest {
+    string instanceId = 1;
+    string name = 2;
+    google.protobuf.StringValue input = 3;
+    bytes guid = 4;
+    bool hasScheduledTime = 5;
```

Comment:
Is this necessary? Can we not test `null` on `scheduledTime`?

Reply:
there is no null for `google.protobuf.Timestamp`. 

That said, I have started using `default` to represent `null` in the queries. I can do that here as well.

Reply:
In C# `Timestamp` from `Google.Protobuf` is a class, so it is nullable.

Reply:
I am a bit concerned about the defaults not matching up (null vs. default(DateTime) vs default-of-Google-Protobuf-Timestamp-in-other-languages) and this coming to bite us later. This is why I had introduced a `bool hasDateTime`. 

We need to make sure I think that we are not using a dotnet-specific convention.

Reply:
> In C# Timestamp from Google.Protobuf is a class, so it is nullable

I find this quite confusing. I am pretty sure that the google timestamp datatype has no null, so I am not sure how this is made sure to round-trip correctly. Maybe a null DateTime would be transmitted as certain Timestamp value and then that certain timestamp value is turned back into null at the other end? But if so, what is that certain value? Is it the default of the DateTime in C#? or the default (two zeros) of Google's internal representation of Timestamp? or whatever other languages (e.g. Java) would consider a default? To save us from having to answer these questions is why I was introducing a `hasTimestamp` field. Unless I get conclusive answers I may have to bring it back.

Reply:
> I am pretty sure that the google timestamp datatype has no null

[Timestamp](https://github.com/protocolbuffers/protobuf/blob/main/csharp/src/Google.Protobuf/WellKnownTypes/TimestampPartial.cs#L16) is a `class`, so it can be `null`.

Reply:
> Timestamp is a class, so it can be null

The point was that *the official Google Timestamp datatype and its protobuf representation have no concept of null*, it really does not matter what is happening in C#. Maybe someone somewhere made a choice as to how the C# null should be represented on the wire. This choice is not a "standard" choice and we need to be aware of its consequences. Have you tested that null roundtrips correctly? And how do we handle this issue when using Java?

Reply:
I am pretty sure that if you set Timestamp to null, send it across the wire, and then read it from the wire, you get a `new Timestamp()` back - not null.

Reply:
I am going to add explicit conversions to make this 100% clear.

Reply:
I just verified - you get `null` on the other end. While protobuf forbids null fields, Timestamp is not a field, it is a message. And messages _can_ be `null` via omission. And this is what we are doing, omitting the `Timestamp` message via `null`.